### PR TITLE
Table: new Table.RowDrawer subcomponent

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

Table: new Table.RowDrawer subcomponent

Screen Shot 2022-11-11 at 12 23 18 PM

Also, removed divider between Row and RowExpandable and RowDrawer

Screen Shot 2022-11-11 at 12 42 55 PM

#### Why?

New component to support m10n requests. 

New featured (Drawer with SlimBanner) was tested under experimentation presenting good results.

### Checklist

- [X] Added unit and Flow Tests
- [X] Added documentation + accessibility tests
- [X] Verified accessibility: keyboard & screen reader interaction
- [X] Checked dark mode, responsiveness, and right-to-left support
- [X] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/pages/web/table.js [hunk 1]
```diff
@@ -1599,6 +1599,16 @@ function Example() {
             id="Table.RowExpandable"
           />
         </MainSection.Subsection>
+        <MainSection.Subsection
+          title={generatedDocGen?.TableRowDrawer?.displayName}
+          description={generatedDocGen?.TableRowDrawer?.description}
+        >
+          <GeneratedPropTable
+            generatedDocGen={generatedDocGen.TableRowDrawer}
+            name="Table.RowDrawer"
```
Comment (dangerismycat) [inline docs/pages/web/table.js:1608]: `name` and `id` can both also use displayName

File: packages/gestalt/src/TableRowDrawer.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * Use [Table.RowDrawer](https://gestalt.pinterest.systems/web/table#Table.RowDrawer) to define a row drawer to display additional content.
+ */
+export default function TableRowDrawer({ children, drawerContents, id }: Props): Node {
+  const { stickyColumns } = useTableContext();
+  const rowRef = useRef();
+  const [columnWidths, setColumnWidths] = useState([]);
+
+  const cs = cx(styles.drawer);
```
Comment (dangerismycat) [inline packages/gestalt/src/TableRowDrawer.js:31]: This isn't necessary — you can pass `styles.drawer` directly where `cs` is used

File: packages/gestalt/src/TableRowExpandable.js [hunk 3]
```diff
@@ -58,7 +58,9 @@ export default function TableRowExpandable({
   hoverStyle = 'gray',
 }: Props): Node {
   const [expanded, setExpanded] = useState(false);
-  const cs = hoverStyle === 'gray' ? cx(styles.hoverShadeGray) : null;
+  const cstr = hoverStyle === 'gray' ? cx(styles.hoverShadeGray) : null;
```
Comment (dangerismycat) [inline packages/gestalt/src/TableRowExpandable.js:61]: `classnames` isn't doing anything here — just use the class names directly where needed

`classnames` is only useful when combining multiple classes, especially if there's conditional logic for which classes are included

**Review by dangerismycat — APPROVED** [review]
Can you add screenshots of the new/changed UI to the PR summary?

File: docs/pages/web/table.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/pages/web/table.js:1608]: DONE applied retroactively

File: packages/gestalt/src/TableRowDrawer.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/TableRowDrawer.js:31]: Didnt know this.
DONE

File: packages/gestalt/src/TableRowExpandable.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/TableRowExpandable.js:61]: DONE

**Pull request merged**